src/sinks/util/tcp.rs

```diff
@@ -191,7 +183,7 @@ impl Sink for TcpSink {
                 //
                 // If this returns `WouldBlock` we know the connection is still
                 // valid and the write will most likely succeed.
-                match conn.read(&mut [0u8; 1]) {
+                match connection.get_mut().read(&mut [0u8; 1]) {
```

Comment:
So before we move forward here I want to ensure we run this in the same reproducible scenario I had before. This if iirc was with sending logs to datadog_logs which uses the tcp sink. I had it hang when submitting a size 1 buffer to check for eof.

Reply:
For reproducing same scenario I'll need datadog account?

Reply:
Yes, but I am mostly leaving this as a note that someone on the team should check before we merge (no need to do it yourself). Likely I will do it since I have it setup.